### PR TITLE
[JENKINS-43507] Update implementation guide

### PR DESCRIPTION
**Pull request by stephenc**
@reviewbybees

### CODE REVIEW

**Comment by stephenc** [comment]
Whitespace diff: https://github.com/jenkinsci/scm-api-plugin/pull/38/files?w=1

File: docs/implementation.adoc [hunk 1]
```diff
@@ -562,6 +568,16 @@ public class MySCMSource extends SCMSource {
 
     // Getters for all the configuration fields
 
+    @NonNull
+    public List<SCMSourceTrait> getTraits() {
+        return new ArrayList<>(traits);
```
Comment (rsandell) [inline docs/implementation.adoc]: `Collections.unmodifiableList(traits)` is what I've seen you use.

File: docs/implementation.adoc [hunk 2]
```diff
@@ -711,8 +727,97 @@ public class MySCMSource extends SCMSource {
...
+    }
+
+    // provide builder-style getters and setters for fields
+
+    @NonNull
+    @Override
+    public MySCM build() {
+        MySCM result = new MySCM(this);
+        result.setHead(head(), revision());
+        // apply the decorations from the fields
```
Comment (rsandell) [inline docs/implementation.adoc]: shouldn't you iterate the traits here and ask them to apply stuff to the scm?

Comment (stephenc) [inline docs/implementation.adoc]: Nope, that is handled by the withRequest() call

File: docs/implementation.adoc (hunk 1 above)
Comment (stephenc) [inline docs/implementation.adoc]: ACK

**Pull request merged**